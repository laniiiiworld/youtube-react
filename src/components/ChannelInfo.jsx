import React from 'react';
import { useQuery } from '@tanstack/react-query';
import { useYoutubeApi } from '../context/YoutubeApiContext';

export default function ChannelInfo({ id, name }) {
  const { youtube } = useYoutubeApi();
  const { data: url } = useQuery(['channel', id], () => youtube.channelImageUrl(id), { staleTime: 1000 * 60 * 5 });

  return (
    <div className='flex items-center my-4 mb-6'>
      {url && <img src={url} alt={name} className='rounded-full w-10 h-10' />}
      <p className='ml-2 font-medium text-lg'>{name}</p>
    </div>
  );
}
