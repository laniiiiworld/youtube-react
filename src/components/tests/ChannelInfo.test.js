import { render, screen, waitFor } from '@testing-library/react';
import { Route } from 'react-router-dom';
import { withAllContexts, withRouter } from '../../tests/utils';
import ChannelInfo from '../ChannelInfo';

describe('ChannelInfo', () => {
  const fakeYoutube = {
    channelImageUrl: jest.fn(),
  };

  afterEach(() => fakeYoutube.channelImageUrl.mockReset());

  it('renders correctly', async () => {
    fakeYoutube.channelImageUrl.mockImplementation(() => 'url');
    const { asFragment } = renderChannerInfo();

    await waitFor(() => screen.getByRole('img'));

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders without URL', () => {
    fakeYoutube.channelImageUrl.mockImplementation(() => {
      throw new Error('error');
    });
    renderChannerInfo();

    expect(screen.queryByRole('img')).toBeNull();
  });

  it('renders with URL', async () => {
    fakeYoutube.channelImageUrl.mockImplementation(() => 'url');
    renderChannerInfo();

    await waitFor(() => expect(screen.getByRole('img')).toBeInTheDocument());
  });

  function renderChannerInfo() {
    return render(
      withAllContexts(
        withRouter(
          <Route path='/' element={<ChannelInfo id='id' name='channel' />} /> //
        ), //
        fakeYoutube
      ) //
    );
  }
});
