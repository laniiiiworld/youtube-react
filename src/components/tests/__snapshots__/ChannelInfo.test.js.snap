// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelInfo renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex items-center my-4 mb-6"
  >
    <img
      alt="channel"
      class="rounded-full w-10 h-10"
      src="url"
    />
    <p
      class="ml-2 font-medium text-lg"
    >
      channel
    </p>
  </div>
</DocumentFragment>
`;
